### DISCUSSION TITLE
Individual artist for each track on compilation

### DISCUSSION THREAD

**Discussion by kenzoJEP** [joalla/discogs_client#133] [Support] [answered]
How can I access each individual artist for each track on various artists' compilations? I can find the position, title, and duration but not the artist(s)

**Comment by AnssiAhola** [comment, marked as answer]
Hey @kenzoJEP 
Could you show the code you are using when trying to access the artist(s) ?

It _should_ work just like this
```python
for track in release.tracklist:
   print(track.artists)
```
There's also `credits` property you can check
```python
print(track.credits) # this is an list of Artist objects
```

Also could you tell us the specific compilation/release you are trying to access?
Could help us track down the issue.

Cheers!
-Anssi

**Comment by kenzoJEP** [comment]
Hi Anssi

Thanks for your reply.

for track in release.tracklist:
    print(track.artists)

Yes, this works as you say it does.

I went wrong as I assigned variables:-

artists = release.artists # [<Artist 194 'Various'>]
track = release.tracklist

for trk in track:
    print(trk.position + ' ' + trk.artists + ' ' + trk.title + ' ' +
trk.duration)

An exception was raised as artists are not in track.

This is the compilation I was using to test:-

https://www.discogs.com/fr/release/2635858-Various-We-Got-A-Party-The-Best-Of-Ron-Records-Volume-One

This leaves me with another problem as trk.artists gives the output of
[<Artist 18394 'Professor Longhair'>]
and I want Professor Longhair. I have tried using

trk_artist = re.findall(r"'([^']*)'", str(trk.artists))

but the output is ['Professor Longhair'] - Close but no cigar.

Do you know of a way to get the desired output?

Thanks,
John

**Reply by AnssiAhola** [reply]
The `artists` property of an track is actually an `ListField` of `Artist` objects, which is basically an array of `Artist` objects.
When you print the `track.artists` it uses the [`__repr__` method](https://github.com/joalla/discogs_client/blob/6292146201aebb49cdef9a639cc1ddf1a5d8941d/discogs_client/models.py#L490-L491) of the `Artist` object to print an string representation of that object.
i.e. Multiple track artists would print out like so
```python
[<Artist 1373771 'Spectral Lore'>, <Artist 3020852 'Mare Cognitum'>]
```
You can access the artist name like so
```python
for track in release.tracklist:
    artist: Artist = track.artists[0] # I added the Artist type here to aid with code completion
    print(track.position + ' ' + artist.name + ' ' + track.title + ' ' + track.duration)
```
_Note: This assumes that `track.artists` always has at least one artist, some guard clauses here might be a good idea_

Output
```
A1 Professor Longhair Go To The Mardi Gras 2:46
A2 Irma Thomas Don't Mess With My Man 2:07
.
.
.
```

Hope that helps!

**Comment by kenzoJEP** [comment]
Hi

Great help - just what I was looking for!

Thanks,
John
